pkg/quota/image/usage.go

```
@@ -10,139 +13,400 @@ import (
+
+	for _, imageStream := range iss.Items {
+		if is != nil && imageStream.Name == is.Name {
+			continue
+		}
+		c.processImageStreamImages(&imageStream, func(_, _ string, img *imageapi.Image) error {
+			processedImages.Insert(img.Name)
+			return nil
+		})
+	}
+
+	if is != nil {
+		c.processImageStreamImages(is, func(_, _ string, img *imageapi.Image) error {
+			if !processedImages.Has(img.Name) {
+				processedImages.Insert(img.Name)
+				imagesIncrement.Set(imagesIncrement.Value() + 1)
+			}
+			return nil
+		})
+	}
+
+	if image != nil && !processedImages.Has(image.Name) {
+		if value, ok := image.Annotations[imageapi.ManagedByOpenShiftAnnotation]; ok && value == "true" {
+			if !processedImages.Has(image.Name) {
+				imagesIncrement.Set(imagesIncrement.Value() + 1)
+			}
+		}
+	}
+
+	images = resource.NewQuantity(int64(len(processedImages)), resource.DecimalSI)
+
+	return
+}
 
-	for _, history := range is.Status.Tags {
+// processImageStreamImages is a utility method that calls a given handler for every image of the given image
+// stream that belongs to internal registry. It process image stream status and optionally spec.
+func (c *GenericImageStreamUsageComputer) processImageStreamImages(
+	is *imageapi.ImageStream,
+	handler func(tag string, dockerImageReference string, image *imageapi.Image) error,
+) error {
+	processedImages := sets.NewString()
+
+	for tag, history := range is.Status.Tags {
 		for i := range history.Items {
 			imageName := history.Items[i].Image
 			if len(history.Items[i].DockerImageReference) == 0 || len(imageName) == 0 {
 				continue
 			}
 
-			if _, exists := processedImages[imageName]; exists {
-				continue
+			var (
+				err error
+				img *imageapi.Image
+			)
+			if c.fetchImagesFromImageStream {
+				img, err = c.getImageStreamImage(is.Namespace, is.Name+"@"+imageName)
+				if err != nil {
+					glog.Errorf("Failed to get image %s of image stream %s/%s: %v", imageName, is.Namespace, is.Name, err)
```

Don't use glog.Errorf() in our code base - if you need to eat an error, use util.HandleError.   However, this is not the right check - you should be specifically ignoring NotFound errors before you log output.